src/Apps/Order/Components/AddressVerificationFlow.tsx

```diff
@@ -100,22 +104,151 @@ const fallbackFromFormValues = (address: AddressValues): AddressOption => {
   }
 }
 
+const useAddressVerificationTracking = () => {
+  const { trackEvent } = useTracking()
+  const { contextPageOwnerSlug } = useAnalyticsContext()
+  const { user } = useSystemContext()
+  const userId = user?.id
+
+  return {
+    trackViewedModal: useCallback(
+      ({ subject, option }: { subject: string; option: string }) => {
+        trackEvent({
+          action_type: "validationAddressViewed",
+          context_module: ContextModule.ordersShipping,
+          context_page_owner_type: OwnerType.ordersShipping,
+          context_page_owner_id: contextPageOwnerSlug,
+          user_id: userId,
+          flow: "user adding shipping address",
+          subject,
+          option,
+        })
+      },
+      [contextPageOwnerSlug, trackEvent, userId]
+    ),
+    trackClickedModal: useCallback(
+      ({
+        label,
+        option,
+        subject = null,
+      }: {
+        label?: string | null
+        option?: string | null
+        subject?: string | null
+      }) => {
+        trackEvent({
+          action_type: "clickedValidationAddress",
+          context_module: ContextModule.ordersShipping,
+          context_page_owner_type: OwnerType.ordersShipping,
+          context_page_owner_id: contextPageOwnerSlug,
+          user_id: userId,
+          ...(typeof subject !== "undefined" && { subject }),
+          ...(typeof option !== "undefined" && { option }),
+          ...(typeof label !== "undefined" && { label }),
+        })
+      },
+      [contextPageOwnerSlug, trackEvent, userId]
+    ),
+  }
+}
+
 const AddressVerificationFlow: React.FC<AddressVerificationFlowProps> = ({
   verificationInput,
   verifyAddress,
   onChosenAddress,
   onClose,
 }) => {
-  const [modalType, setModalType] = useState<ModalType | null>(null)
-  const [addressOptions, setAddressOptions] = useState<AddressOption[]>([])
+  const [showModal, setShowModal] = useState(false)
+  const didLoad = useRef(false)
+
+  let modalType: FlowType = FlowType.REVIEW_AND_CONFIRM
+  let addressOptions: AddressOption[] = []
+
+  const {
+    trackViewedModal,
+    trackClickedModal,
+  } = useAddressVerificationTracking()
+
+  const error = (verifyAddress.verifyAddressOrError as VerifyAddressErrorType)
+    ?.mutationError
+  const suggestedAddresses = (verifyAddress.verifyAddressOrError as VerifyAddressSuccessType)
+    ?.suggestedAddresses
+  const inputAddress = (verifyAddress.verifyAddressOrError as VerifyAddressSuccessType)
+    ?.inputAddress
+  const verificationStatus = (verifyAddress.verifyAddressOrError as VerifyAddressSuccessType)
+    ?.verificationStatus
+
+  const hasError = Boolean(error)
+
+  if (hasError) {
```

Comment:
Previously this whole section ran in a hook, but it depends on props which don't currently change after load. If we changed the behavior to run multiple verifications on a flow then our use of the `didLoad` ref would have to change to allow updates, but this code should now be stable.